### PR TITLE
Added solutions to strings problems

### LINKED ISSUE #1938: Incomplete resources for sample solutions to strings code challenges
### Page where problem found?

(https://github.com/Techtonica/curriculum/blob/main/code-challenges/string.md)

### Type of problem

 Incomplete resources for sample solutions to strings code challenges

### Suggested Solution

_Optional. If you would like to suggest a specific solution, please add it here._

### PR DESCRIPTION
**Pull request by archijaiswal**
Fixes #1938

### CODE REVIEW

File: code-challenges/string.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ s2 = "eygoodb"
 Expected output: False
 
 Sample Solution:
-
+```
```
Comment (allejo) [inline code-challenges/string.md:68]:
~~~suggestion
```javascript
~~~

You can specify what language this code is and it'll add syntax highlighting to it! Please this to all the code blocks you've touched

File: code-challenges/string.md [hunk 2]
```diff
@@ -107,8 +124,25 @@ front_times('Chocolate', 2) → 'ChoCho'
...
 front_times('Ok', 4) → 'OkOkOkOk'
 
-Needs sample solution
+Sample solution
+```
+function front_times(str, n) {
 
+  var front = str.substring(0, 3);
+  var result = "";
+  for (var i = 0; i < n; i++) {
```
Comment (allejo) [inline code-challenges/string.md:133]:
```suggestion
  const front = str.substring(0, 3);
  let result = "";
  for (let i = 0; i < n; i++) {
```

Please use `const` or `let` instead of `var`.

https://medium.com/@codingsam/awesome-javascript-no-more-var-working-title-999428999994

tl;dr:

- Use `const` when a variable won't be reassigned (i.e. you won't use an equal sign on it again)
- Use `let` when you need to reassign a value (i.e. you'll use an equal sign on it in the future)

File: code-challenges/string.md [hunk 3]
```diff
@@ -176,3 +327,20 @@ Given an "out" string length 4, such as "<<>>", and a word, return a new string
 make_out_word('<<>>', 'Yay') → '<<Yay>>'
```
Comment (allejo) [inline code-challenges/string.md:327]: @Yosolita1978, for this question, is `out` always going to be of length 4? or should this solution also handle something like `##[]##`?

**Review by allejo — CHANGES_REQUESTED** [review]

File: code-challenges/string.md (hunk 1 above)
Comment (archijaiswal) [reply code-challenges/string.md:68]: Sure, thanks!

File: code-challenges/string.md (hunk 3 above)
Comment (Yosolita1978) [reply code-challenges/string.md:327]: For this problem, you can assume that `out` will always be length 4.
